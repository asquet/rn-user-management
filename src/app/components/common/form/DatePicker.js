import React from 'react';
import ForeignDatePicker from 'react-native-datepicker';
import { View } from 'react-native';
import { formStyles } from '../../../stylesheet';

export default function DatePicker({ style, ...props }) {
  return (
    <ForeignDatePicker
      style={[formStyles.datePicker, style]}
      customStyles={{
        dateIcon: formStyles.datePickerIcon,
        dateInput: formStyles.datePickerInput,
        dateText: formStyles.datePickerText,
      }}
      {...props}
    />
  );
}

DatePicker.propTypes = {
  style: View.propTypes.style,
};

DatePicker.defaultProps = {
  style: null,
};
