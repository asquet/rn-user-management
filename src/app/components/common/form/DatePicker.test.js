import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import DatePicker from './DatePicker';

// Note: test renderer must be required after react-native.

describe('DatePicker', () => {

  it('renders correctly', () => {
    const date = new Date(123123);

    const tree = renderer.create(
      <DatePicker date={date} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
