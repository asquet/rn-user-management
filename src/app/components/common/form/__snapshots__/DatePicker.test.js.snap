// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "width": 142,
        },
        Object {
          "flex": 1,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#aaa",
            "borderWidth": 1,
            "flex": 1,
            "height": 40,
            "justifyContent": "center",
          },
          undefined,
          false,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333",
            },
            undefined,
          ]
        }
      >
        1970-01-01
      </Text>
    </View>
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 32,
            "marginLeft": 5,
            "marginRight": 5,
            "width": 32,
          },
          undefined,
        ]
      }
    />
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#00000077",
                "flex": 1,
                "flexDirection": "row",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#fff",
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderTopColor": "#ccc",
                      "borderTopWidth": 1,
                      "marginTop": 42,
                    },
                    undefined,
                  ]
                }
              >
                <RCTDatePicker
                  date={123123}
                  maximumDate={undefined}
                  minimumDate={undefined}
                  minuteInterval={undefined}
                  mode="date"
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "height": 216,
                    }
                  }
                  timeZoneOffsetInMinutes={undefined}
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "padding": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "left": 0,
                      },
                      undefined,
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#46cf98",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#666",
                      },
                      undefined,
                    ]
                  }
                >
                  取消
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "padding": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "right": 0,
                      },
                      undefined,
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#46cf98",
                        "fontSize": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  确定
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
